### PR TITLE
feat(core): get /users/:userId/logs

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Get user logs from `GET /users/:userId/logs`

## Linear Issue Reference

LOG-32

## Testing

image

### CODE REVIEW

File: packages/core/src/queries/log.ts [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+    from ${table}
+    where ${fields.payload}->>'userId'=${userId}
+    ${conditionalSql(
+      applicationId,
+      (applicationId) => sql`and ${fields.payload}->>'applicationId'=${applicationId}`
+    )}
+    ${conditionalSql(logType, (logType) => sql`and ${fields.type}=${logType}`)}
+    limit ${limit}
+    offset ${offset}
+  `);
```
Comment (demonzoo) [inline packages/core/src/queries/log.ts:46]: Add unit tests for this file pls

Comment (IceHe) [reply packages/core/src/queries/log.ts:46]: There is no need to test slonik `sql` for now.

**Comment by IceHe** [comment]
Rebased master

File: packages/core/src/queries/log.ts [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+
+export const findUserLogs = async (
+  limit: number,
+  offset: number,
+  { userId, applicationId, logType }: UserLogCondition
+) =>
+  envSet.pool.any<Log>(sql`
+    select ${sql.join(Object.values(fields), sql`,`)}
+    from ${table}
+    where ${fields.payload}->>'userId'=${userId}
```
Comment (demonzoo) [inline packages/core/src/queries/log.ts:38]: What is this `->>` arrow?

**Review by demonzoo — APPROVED** [review]

File: packages/core/src/queries/log.ts (hunk 2 above)
Comment (IceHe) [reply packages/core/src/queries/log.ts:38]: access the data in jsonb field

Comment (demonzoo) [reply packages/core/src/queries/log.ts:38]: Yeah, I get it now. Just checked postgresql document

**Comment by IceHe** [comment]
Use `GET /logs?userId=:userId` instead

**Pull request closed without merging**